File: app/src/main/java/com/demo/room/bean/Person.java
package com.demo.room.bean;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "t_person")
public class Person {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "p_id")
    public int pid;

    /*不修改字段名就不需要加@ColumnInfo注解*/
    public String name;

    public int age;

    @Embedded(prefix = "one")/*实体类中引用其他实体类*/
    public Address address1;
    @Embedded(prefix = "two")/*实体类中引用其他实体类*/
    public Address address2;

    @Ignore
    public float income;
//    /**
//     * 新加字段,用于演示数据库升级
//     */
//    public String phone;

    public Person(String name, int age) {
        this.name = name;
        this.age = age;
    }

    @NonNull
    @Override
    public String toString() {
        return "[" + pid + ", " +
                name + ", " +
                age + "岁, " +
                income + "元" +
                (address1 == null ? ", " : address1.toString()) +
                (address2 == null ? ", " : address2.toString()) + "]";
    }

    public static class Address {
        public String city;
        public String street;

        public Address(String city, String street) {
            this.city = city;
            this.street = street;
        }

        @NonNull
        @Override
        public String toString() {
            return ", " + city + ", " + street;
        }
    }
}

File: app/src/main/java/com/gxd/demo/room/table/Person.kt
package com.gxd.demo.room.table

import androidx.room.*

/**
 * 如果您的应用需要通过全文搜索 (FTS) 快速访问数据库信息，请使用虚拟表（使用 FTS3 或 FTS4 SQLite 扩展模块）为您的实体提供支持
 */
@Entity(tableName = "t_person")
data class Person
constructor(
    /**
     * SQLite中的表和列名称不区分大小写
     * 使用全文搜索时，铸剑名称必须为p_id且类型为INTEGER
     */
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "p_id")
    val id: Int = 0,
    /**
     * 不修改字段名就不需要加@ColumnInfo注解
     */
    val name: String,
    val age: Int,
    /**
     * 实体类中引用其他实体类
     */
    @Embedded(prefix = "first_")
    val firstAddress: Address,

    @Embedded(prefix = "second_")
    val secondAddress: Address? = null
) {
    /**
     * Ignore表示该字段不存入表中
     */
    @Ignore
    val income: Int? = null
}

data class Address(
    val country: String,
    val city: String? = null,
)